packages/paymasters/contracts/TokenPaymaster.sol

```diff
@@ -102,7 +96,11 @@ contract TokenPaymaster is BasePaymaster {
     virtual
     relayHubOnly
     returns (bytes memory context, bool revertOnRecipientRevert) {
-        (relayRequest, signature, approvalData, maxPossibleGas);
+        (signature);
+
+        require(approvalData.length == 0, "invalid approval data length");
+        require(relayRequest.relayData.paymasterData.length == 32, "invalid paymaster data length");
```

Comment:
here its even worse: `invalid paymaster data length` give no clue as to what it should contain (in this case, a uniswap address, which is used in another place (in _getToken)
please restore the meaningful error messages.